packages/amplify-category-function/src/provider-utils/awscloudformation/utils/storeResources.ts

```diff
@@ -102,6 +102,25 @@ export function createLayerCfnFile(context, parameters, layerDirPath) {
   });
 }
 
+export function updateLayerCfnFile(context, parameters, layerDirPath) {
+  context.amplify.writeObjectAsJson(
+    path.join(layerDirPath, parameters.layerName + '-awscloudformation-template.json'),
+    generateLayerCfnObj(parameters),
+    true,
+  );
+  context.amplify.updateamplifyMetaAfterResourceUpdate(categoryName, parameters.layerName,'runtimes', parameters.runtimes);
+
+}
+
+export function createLayerParametersFile(context,parameters,layerDirPath){
+  let parametersFileName = 'layer-parameters.json';
+  fs.ensureDirSync(layerDirPath);
+  const parametersFilePath = path.join(layerDirPath, parametersFileName);
+  const currentParameters = fs.existsSync(parametersFilePath) ? context.amplify.readJsonFile(parametersFilePath) : {};
+  const jsonString = JSON.stringify({ ...currentParameters, ...parameters }, null, 4);
```

Comment:
does this spread operation work if someone removes all of the layerPermissions?

Reply:
Ohh Nice catch , fixed